Engine/SpacePartition/OLAABBTree.cpp

```diff
@@ -0,0 +1,441 @@
+#include "OLAABBTree.h"
+#include <assert.h>
+#include <stack>
+#include "UI/DebugDraw.h"
+
+
+OLAABBTree::OLAABBTree(unsigned initial_size)
+{
+	node_capacity = initial_size;
+	//Resize
+	nodes.resize(initial_size);
+	for (unsigned node_index = 0; node_index < initial_size; node_index++)
```

Comment:
In the for loop I recommend to do '++node_index' because when doing 'node_index++' you make a copy and then add to it, instead with '++node_index' you directly increment the value. More efficient tip! :)